game-core/src/main/java/games/strategy/net/ServerMessenger.java

```diff
@@ -166,16 +165,12 @@ private boolean isGame() {
   }
 
   private final Object cachedListLock = new Object();
-  private final HashMap<String, String> cachedMacAddresses = new HashMap<>();
+  private final Map<String, String> cachedMacAddresses = new HashMap<>();
 
   @Override
   public @Nullable String getPlayerMac(final String name) {
     synchronized (cachedListLock) {
-      String mac = cachedMacAddresses.get(name);
-      if (mac == null) {
-        mac = playersThatLeftMacsLast10.get(name);
-      }
-      return mac;
+      return cachedMacAddresses.getOrDefault(name, playersThatLeftMacsLast10.get(name));
```

Comment:
The JDK really needs an overload of `Map#getOrDefault()` that accepts a `Supplier` so we're not forced to do a lookup for the default value that may not actually be needed.  Using an `Optional` may be overkill here:

```java
return Optional.ofNullable(cachedMacAddresses.get(name))
    .orElseGet(() -> playersThatLeftMacsLast10.get(name));
```